sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InMemoryFileIndex.scala

```diff
@@ -290,6 +291,7 @@ object InMemoryFileIndex extends Logging {
       isRootPath: Boolean): Seq[FileStatus] = {
     logTrace(s"Listing $path")
     val fs = path.getFileSystem(hadoopConf)
+    val ignoreLocality = hadoopConf.getBoolean(SQLConf.IGNORE_DATA_LOCALITY.key, false)
```

Comment:
Could we move this config to [InMemoryFileIndex.scala#L198](https://github.com/apache/spark/blob/0da290368ae7bf59aa37b3a1aac514808510f484/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InMemoryFileIndex.scala#L198).

Reply:
Fixed.